core/blockchain.go

```diff
@@ -1382,6 +1384,18 @@ func (bc *BlockChain) insertBlock(block *types.Block, writes bool) error {
 	triedbCommitTimer.Inc(statedb.TrieDBCommits.Milliseconds())     // Triedb commits are complete, we can mark them
 	blockWriteTimer.Inc((time.Since(substart) - statedb.AccountCommits - statedb.StorageCommits - statedb.SnapshotCommits - statedb.TrieDBCommits).Milliseconds())
 	blockInsertTimer.Inc(time.Since(start).Milliseconds())
+	if baseFee := block.BaseFee(); baseFee != nil {
+		blockGasPriceGauge.Update(baseFee.Int64())
+	}
+
+	totalFees := big.NewInt(0)
+	for _, tx := range block.Transactions() {
+		gasPrice := tx.GasPrice()
+		gasUsed := new(big.Int).SetUint64(tx.Gas())
+		txFees := new(big.Int).Mul(gasPrice, gasUsed)
+		totalFees = new(big.Int).Add(totalFees, txFees)
+	}
+	blockTotalFeesGauge.Update(totalFees.Int64())
```

Comment:
Can we move these metrics into Accept here: https://github.com/ava-labs/subnet-evm/blob/master/core/blockchain.go#L1055 ? 

We should also update the `totalFees` calculation to account for the base fee and effective gas price. Unfortunately, `GasPrice` got a bit more complicated with EIP-1559: https://timroughgarden.org/papers/eip1559.pdf.

The transaction's `GasPrice()` function returns the gas price for legacy transactions (https://github.com/ava-labs/subnet-evm/blob/master/core/types/legacy_tx.go#L109) and the `GasFeeCap` for transactions added post-EIP1559 (https://github.com/ava-labs/subnet-evm/blob/master/core/types/dynamic_fee_tx.go#L102), which does not account for the actual fee paid anymore.

We actually do this incorrectly in the logs here too: https://github.com/ava-labs/subnet-evm/blob/master/miner/worker.go#L391, so I think we can leave this in this PR for now and make a ticket to correct this and factor out the code so that it can be shared across these two places.

Reply:
Thanks for the feedback @aaronbuchwald, I moved the changes to the place you suggested I also update the total Fees calculation to include `GasFeeCap()` and `GasTipCap()`.